clientapi/routing/auth_fallback.go

```diff
@@ -102,6 +103,22 @@ func AuthFallback(
 	w http.ResponseWriter, req *http.Request, authType string,
 	cfg *config.ClientAPI,
 ) *util.JSONResponse {
+	// We currently only support reCaptcha, so fail early if that's not requested
+	if authType == authtypes.LoginTypeRecaptcha {
+		if !cfg.RecaptchaEnabled {
+			return writeHTTPMessage(w, req,
+				"Recaptcha login is disabled on this Homeserver",
+				http.StatusBadRequest,
+			)
+		}
+	} else {
+		_ = writeHTTPMessage(w, req, fmt.Sprintf("Unknown authtype %q", authType), http.StatusNotImplemented)
+		return &util.JSONResponse{
+			Code: http.StatusNotFound,
+			JSON: jsonerror.NotFound("Unknown auth stage type"),
```

Comment:
These JSON responses are presumably just logged? Given we write back text on `:115`? Needs clarification / removal of `util.JSONResponse` here.

Reply:
x100 on this. It's currently really unclear when it is right to send back text, and when to send back JSON, and the whole interplay between the two. Like here, it's an error and we write back text. But if we fail to parse the form values we send back JSON. Why?

Reply:
TBH, I also don't like all the JSON and normal HTML stuff combined. Since we're just using the `MakeHTMLAPI` for auth fallback right now, I'd be totally fine to completely remove JSON here.